### ISSUE TITLE
Mac version of Kiwix Hotspot cannot see SD card

### ISSUE THREAD

**Issue by rgaudin** [kiwix/kiwix-hotspot#389]
In GitLab by @mrjohnc on Jan 16, 2019, 14:16

I've tried several micro SD cards and formatted them to several different formats but I still get the same issue, the system can see and write to the micro SD card but Kiwix Hotspot cannot see it. I have tried refreshing, restarting the computer, tried different ports (bot SD card slot and USB card reader).

![Screen_Shot_2019-01-16_at_15.11.25](/uploads/952c66fa2e621279cfc7b678a87d494f/Screen_Shot_2019-01-16_at_15.11.25.png)

**Comment by rgaudin** [comment]
In GitLab by @rgaudin on Jan 16, 2019, 16:35

Thanks @mrjohnc,

This list is populated with Disk entries from the operating system with two conditions:

* It detected a non-zero size (seems to be the case here)
* It is marked as *removable* (not an internal disk).

The icon in your screenshot made me think it is maybe detected as an internal disk but on my computer, when using the built-in card reader, I get the same icon.

So can you please send us the output of the `diskutil list` command ? (With the card inserted obviously).

Mine is:

```
/dev/disk0 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *500.3 GB   disk0
   1:                        EFI EFI                     209.7 MB   disk0s1
   2:                 Apple_APFS Container disk1         500.1 GB   disk0s2

/dev/disk1 (synthesized):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      APFS Container Scheme -                      +500.1 GB   disk1
                                 Physical Store disk0s2
   1:                APFS Volume Œuf                     450.6 GB   disk1s1
   2:                APFS Volume Preboot                 42.0 MB    disk1s2
   3:                APFS Volume Recovery                517.1 MB   disk1s3
   4:                APFS Volume VM                      7.5 GB     disk1s4

/dev/disk5 (external, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:                                                   *63.9 GB    disk5
```

The important part is at the bottom, where it says **external**.

Bellow are screenshots of *Disk Utility* and Kiwix Hotspot with both the internal reader and an external USB reader.

![DU-USB](/uploads/68a2a1b43c78e56f14c8211c070422b2/DU-USB.png)

![KH-USB](/uploads/e6e217f3e68b0b7463c108f9e89cb3f3/KH-USB.png)

![KH-builtin](/uploads/dd7a8880a5f392ee2dc30f92bb09d365/KH-builtin.png)

![DU-builtin](/uploads/55fd5be5557bd8e254caa781b3da48e2/DU-builtin.png)

**Comment by rgaudin** [comment]
In GitLab by @mrjohnc on Jan 18, 2019, 10:47

OK, weird thing happened... I tried a smaller SD card on the same computer and it sees it. Here's the output for the smaller SD card

```
mac-74165:~ johncummings$ diskutil list
/dev/disk0 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *251.0 GB   disk0
   1:                        EFI EFI                     209.7 MB   disk0s1
   2:                  Apple_HFS ERI HD                  170.8 GB   disk0s2
   3:                 Apple_Boot Recovery HD             650.0 MB   disk0s3
   4:                  Apple_HFS Micro SD                79.2 GB    disk0s4

/dev/disk1 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:     FDisk_partition_scheme                        *7.9 GB     disk1
   1:                 DOS_FAT_32 NO NAME                 7.9 GB     disk1s1
```

And Disk Utility says

```
Volume type : Physical Device
BSD device node : disk1
Connection : USB
Device tree path : IODeviceTree:/PCI0@0/EHC2@1A,7
Writable : No
Is case-sensitive : No
Volume capacity : 7,948,206,080
Available space (Purgeable + Free) : 0
Purgeable space : 0
Free space : 0
Used space : 7,948,206,080
Owners enabled : No
Is encrypted : No
Can be verified : No
Can be repaired : No
Bootable : No
Journaled : No
Disk number : 1
Media name : APPLE SD Card Reader Media
Media type : Generic
Ejectable : Yes
Solid state : No
S.M.A.R.T. status : Not Supported
```

I know the larger SD card (128GB) is not broken because I now have it running Kiwix on my Raspberry Pi, I'll send the info for that one later today.

**Comment by rgaudin** [comment]
In GitLab by @mrjohnc on Jan 18, 2019, 13:32

Now I'm very confused, this is the same SD card I had the issue with showing up in Kiwix Hotspot fine. The only difference is I installed Kiwix on it from another computer. I have no idea.....Maybe a card formatting issue? I notice the 128GB SD card has the wrong capacity in the first image so maybe a failed install (the failed install from issue 299) does something odd with partitions that normal Disk Utility cannot see but Windows 10 can? 

![Screen_Shot_2019-01-18_at_14.12.45](/uploads/c424a1909cc6453e95b2e07ec2dfa6ef/Screen_Shot_2019-01-18_at_14.12.45.png)

**Comment by rgaudin** [comment]
In GitLab by @rgaudin on Feb 12, 2019, 10:49

closed via commit 81c91fa4688d6738ec44927e1106de64c3a6916b

**Comment by rgaudin** [comment]
In GitLab by @rgaudin on Jun 25, 2019, 09:46

mentioned in commit 81c91fa4688d6738ec44927e1106de64c3a6916b
